physics/MP/GFDL/v3_2022/gfdl_cloud_microphys_v3_mod.F90

```diff
@@ -0,0 +1,7377 @@
+            call mtetw (ks, ke, qv (i, :), ql (i, :), qr (i, :), qi (i, :), &
+                qs (i, :), qg (i, :), tz, ua (i, :), va (i, :), wa (i, :), &
+                delp (i, :), dte (i), 0.0, water (i), rain (i), ice (i), &
+                snow (i), graupel (i), 0.0, 0.0, dtm, te_end_m (i, :), &
+                tw_end_m (i, :), te_b_end_m (i), tw_b_end_m (i), .true., hydrostatic)
+        endif
+
+        ! -----------------------------------------------------------------------
+        ! calculate total energy loss or gain
+        ! -----------------------------------------------------------------------
+
+        if (consv_te) then
+            if (hydrostatic) then
+                do k = ks, ke
+                    te (i, k) = te (i, k) + c_air * tz (k) * delp (i, k)
+                enddo
+            else
+                do k = ks, ke
+                    te (i, k) = te (i, k) + mte (qv (i, k), ql (i, k), qr (i, k), qi (i, k), &
+                        qs (i, k), qg (i, k), tz (k), delp (i, k), .true.) * grav
+                enddo
+            endif
+        endif
+
+        ! -----------------------------------------------------------------------
+        ! conversion of temperature
+        ! -----------------------------------------------------------------------
+
+        if (do_inline_mp) then
+            do k = ks, ke
+                q_cond = qlz (k) + qrz (k) + qiz (k) + qsz (k) + qgz (k)
+                if (cp_heating) then
+                    con_r8 = one_r8 - (qvz (k) + q_cond)
+                    c8 = mhc (con_r8, qvz (k), q_liq (k), q_sol (k)) * c_air
+                    cp8 = con_r8 * cp_air + qvz (k) * cp_vap + q_liq (k) * c_liq + q_sol (k) * c_ice
+                    delz (i, k) = delz (i, k) / pt (i, k)
+                    pt (i, k) = pt (i, k) + (tz (k) * ((1. + zvir * qvz (k)) * (1. - q_cond)) - pt (i, k)) * c8 / cp8
+                    delz (i, k) = delz (i, k) * pt (i, k)
+                else
+                    pt (i, k) = tz (k) * ((1. + zvir * qvz (k)) * (1. - q_cond))
+                endif
+            enddo
+        else
+            do k = ks, ke
+                q_liq (k) = qlz (k) + qrz (k)
+                q_sol (k) = qiz (k) + qsz (k) + qgz (k)
+                q_cond = q_liq (k) + q_sol (k)
+                con_r8 = one_r8 - (qvz (k) + q_cond)
+                c8 = mhc (con_r8, qvz (k), q_liq (k), q_sol (k)) * c_air
+                pt (i, k) = pt (i, k) + (tz (k) - pt (i, k)) * c8 / cp_air
+            enddo
+        endif
+
+        ! -----------------------------------------------------------------------
+        ! total energy checker
+        ! -----------------------------------------------------------------------
+
+        if (consv_checker) then
+            if (abs (sum (te_end_d (i, :)) + te_b_end_d (i) - sum (te_beg_d (i, :)) - te_b_beg_d (i)) / &
+                 (sum (te_beg_d (i, :)) + te_b_beg_d (i)) .gt. te_err) then
+                print*, "GFDL-MP-DRY TE: ", &
```

Comment:
Consider wrapping print statements with a debug flag to reduce standard out spam for normal execution.